pkg/pool-utils/test/LegacyBasePool.test.ts

```diff
@@ -287,113 +287,248 @@ describe('LegacyBasePool', function () {
+          pool = await deployBasePool({
+            pauseWindowDuration: PAUSE_WINDOW_DURATION,
+            bufferPeriodDuration: BUFFER_PERIOD_DURATION,
+            owner,
+          });
+        });
+
+        beforeEach('set sender', () => {
+          sender = other;
+        });
+
+        context('when the sender does not have the pause permission in the authorizer', () => {
+          itRevertsWithUnallowedSender();
+        });
+
+        context('when the sender has the pause permission in the authorizer', () => {
+          sharedBeforeEach('grant permission', async () => {
+            const pauseAction = await actionId(pool, 'pause');
+            const unpauseAction = await actionId(pool, 'unpause');
+            await authorizer
+              .connect(admin)
+              .grantPermissions([pauseAction, unpauseAction], sender.address, [ANY_ADDRESS, ANY_ADDRESS]);
+          });
+
+          itCanPause();
+        });
       });
 
-      context('when the sender has the pause permission in the authorizer', () => {
-        sharedBeforeEach('grant permission', async () => {
-          const pauseAction = await actionId(pool, 'pause');
-          const unpauseAction = await actionId(pool, 'unpause');
-          await authorizer
-            .connect(admin)
-            .grantPermissions([pauseAction, unpauseAction], sender.address, [ANY_ADDRESS, ANY_ADDRESS]);
+      context('with an owner', () => {
+        let owner: SignerWithAddress;
+
+        sharedBeforeEach('deploy pool', async () => {
+          owner = poolOwner;
+          pool = await deployBasePool({
+            pauseWindowDuration: PAUSE_WINDOW_DURATION,
+            bufferPeriodDuration: BUFFER_PERIOD_DURATION,
+            owner,
+          });
         });
 
-        itCanPause();
+        context('when the sender is the owner', () => {
+          beforeEach('set sender', () => {
+            sender = owner;
+          });
+
+          itRevertsWithUnallowedSender();
+        });
+
+        context('when the sender is not the owner', () => {
+          beforeEach('set sender', () => {
+            sender = other;
+          });
+
+          context('when the sender does not have the pause permission in the authorizer', () => {
+            itRevertsWithUnallowedSender();
+          });
+
+          context('when the sender has the pause permission in the authorizer', () => {
+            sharedBeforeEach(async () => {
+              const pauseAction = await actionId(pool, 'pause');
+              const unpauseAction = await actionId(pool, 'unpause');
+              await authorizer
+                .connect(admin)
+                .grantPermissions([pauseAction, unpauseAction], sender.address, [ANY_ADDRESS, ANY_ADDRESS]);
+            });
+
+            itCanPause();
+          });
+        });
       });
     });
 
-    context('with an owner', () => {
-      let owner: SignerWithAddress;
+    describe('recovery mode', () => {
```

Comment:
These tests are good, but they are recovery mode tests, not LegacyBasePool tests. We should:

 - add tests for `_doRecoveryModeExit` (expose in an external function in a MockRecoveryMode contract, call with values and check that we get the proportional amounts out rounded down)
 -  move these testst to a recovery mode test file
 - add tests for the legacy base pool where we have it enter recovery mode and a recovery mode exit is performed via the vault, burning BPT etc